main/src/main/java/org/mobilitydata/gtfsvalidator/validator/DuplicateRouteNameValidator.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Google LLC, MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.annotation.Inject;
+import org.mobilitydata.gtfsvalidator.notice.DuplicateRouteNameNotice;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsRoute;
+import org.mobilitydata.gtfsvalidator.table.GtfsRouteTableContainer;
+
+/**
+ * Validates unicity of short and long name for all routes.
+ *
+ * <p>Generated notice:
+ *
+ * <ul>
+ *   <li>{@link DuplicateRouteNameNotice}
+ * </ul>
+ */
+@GtfsValidator
+public class DuplicateRouteNameValidator extends FileValidator {
+  @Inject GtfsRouteTableContainer routeTable;
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    final Map<String, GtfsRoute> routeByRouteLongName = new HashMap<>(routeTable.entityCount());
+    final Map<String, GtfsRoute> routeByShortName = new HashMap<>(routeTable.entityCount());
+    routeTable
+        .getEntities()
+        .forEach(
+            route -> {
+              if (route.hasRouteLongName()) {
+                if (routeByRouteLongName.containsKey(route.routeLongName())) {
+                  if (areRoutesFromSameAgency(
+                      route.agencyId(),
+                      routeByRouteLongName.get(route.routeLongName()).agencyId())) {
```

Comment:
Do you mean just using `.get()` that will return `null` if the given key a key of the HashMap?

Reply:
I think @aababilov is saying that `.containsKey()` and `.get()` both require a hash table lookup. 

`.containsKey()` can be functionally equivalent as checking if `.get()` returns `null`, because `.get()` returns `null` if the value isn't in the hash table. 

So you can reduce this to a single lookup by calling `.get()` initially and checking if it's null, and if not then using the value that was already returned by `.get()`.

So something like:

```java
                GtfsRoute cachedRoute = routeByLongName.get(route.routeLongName());
                if (cachedRoute != null) {
                  if (areRoutesFromSameAgency(
                      route.agencyId(), cachedRoute.agencyId())) {
```

Reply:
Thanks @barbeau !